wagtail/contrib/redirects/views.py

```diff
@@ -39,63 +40,83 @@
 permission_checker = PermissionPolicyChecker(permission_policy)
 
 
-@permission_checker.require_any("add", "change", "delete")
-@vary_on_headers("X-Requested-With")
-def index(request):
-    query_string = request.GET.get("q", "")
-    ordering = request.GET.get("ordering", "old_path")
-
-    redirects = models.Redirect.objects.prefetch_related("redirect_page", "site")
+class RedirectTargetColumn(Column):
+    cell_template_name = "wagtailredirects/redirect_target_cell.html"
 
-    # Search
-    if query_string:
-        redirects = redirects.filter(
-            Q(old_path__icontains=query_string)
-            | Q(redirect_page__url_path__icontains=query_string)
-            | Q(redirect_link__icontains=query_string)
-        )
 
-    # Ordering (A bit useless at the moment as only 'old_path' is allowed)
-    if ordering not in ["old_path"]:
-        ordering = "old_path"
+class IndexView(generic.IndexView):
+    template_name = "wagtailredirects/index.html"
+    results_template_name = "wagtailredirects/index_results.html"
+    permission_policy = permission_policy
+    model = Redirect
+    header_icon = "redirect"
+    add_item_label = _("Add redirect")
+    context_object_name = "redirects"
+    index_url_name = "wagtailredirects:index"
+    add_url_name = "wagtailredirects:add"
+    edit_url_name = "wagtailredirects:edit"
+    delete_url_name = "wagtailredirects:delete"
+    index_results_url_name = "wagtailredirects:index_results"
+    default_ordering = "old_path"
+    paginate_by = 20
+    is_searchable = True
+    page_title = gettext_lazy("Redirects")
+    search_fields = ["old_path", "redirect_page__url_path", "redirect_link"]
+    _show_breadcrumbs = True
+    columns = [
+        TitleColumn(
+            "old_path",
+            label=gettext_lazy("From"),
+            width="35%",
+            url_name="wagtailredirects:edit",
+            sort_key="old_path",
+        ),
+        Column(
+            "site",
+            label=gettext_lazy("Site"),
+            width="25%",
+            sort_key="site",
```

Comment:
This would be more useful if we use the site's `site_name` field, so it doesn't order by the site's primary key.

```suggestion
            sort_key="site__site_name",
```